### DISCUSSION TITLE
Seeking guidance on using PostMarshallHttpAuthenticationFeature with Minimal API

### DISCUSSION THREAD

**Discussion by AdalbertKhoolsaat** [aws/aws-lambda-dotnet#1679] [Q&A] [answered]
Grateful for the release of the .NET 8 managed runtime for Lambda, I set out to upgrade a number of our Lambdas. 
One particular Web API (originally built in .NET 3.1) was set up with a custom implementation of the `PostMarshallHttpAuthenticationFeature`-override method on the `LambdaEntryPoint`-class, to allow proper interpretation of incoming `APIGatewayCustomAuthorizerContext`.

However, it is unclear to me how to migrate this logic over to a Minimal API setup. 

The underlying (base) classes are ultimately the same, but I can't find out how I am supposed to use this override method, and whether or not this is even intended in a Minimal API setting. 

Any suggestions are welcome!

**Comment by ashishdhingra** [comment]
@AdalbertKhoolsaat Looking at the `AddAWSLambdaHosting` extension methods in [ServiceCollectionExtensions](https://github.com/aws/aws-lambda-dotnet/blob/8435732d39362257d39e5c8556e975a7acd632d9/Libraries/src/Amazon.Lambda.AspNetCoreServer.Hosting/ServiceCollectionExtensions.cs#L36) class, I don't see any option in `HostingOptions` to specify `PostMarshallHttpAuthenticationFeature` override.

CC @normj for inputs if this is a feature request.

**Comment by normj** [comment, marked as answer]
@ashishdhingra is correct, we haven't provided a mechanism yet for overriding the same `PostX` methods that are overridable when using the non-minimal api approach. We need to figure a pattern for users to inject their lambda function sub class instead of using the enum for the AddAWSLamdbaHosting.

**Comment by heathprovost** [comment]
This is a blocker for our migration to minimal API as well for a bunch of our lambdas.
